tests/integration/buildcmd/test_build_cmd.py

```diff
@@ -1691,7 +1694,59 @@ def test_build_layer_with_architecture_not_compatible(self, build_method, use_co
         command_result = run_command(cmdlist, cwd=self.working_dir)
         # Capture warning
         self.assertIn(
-            f"Layer `{layer_identifier}` has BuildArchitecture `x86_64`, which is not listed in CompatibleArchitectures.",
+            f"Layer '{layer_identifier}' has BuildArchitecture x86_64, which is not listed in CompatibleArchitectures",
+            str(command_result.stderr.decode("utf-8")),
+        )
+        # Build should still succeed
+        self.assertEqual(command_result.process.returncode, 0)
+
+    @parameterized.expand([("python3.8", False), ("python3.8", "use_container")])
+    def test_build_arch_no_compatible_arch(self, runtime, use_container):
```

Comment:
nit: AFAIU, these tests are not validating the case where there is no CompatibleArchitectures or no BuildArchitecture present, can we add some tests for them (could be unit tests)?

Reply:
I believe the[ LayerOne build tests ](https://github.com/aws/aws-sam-cli/blob/1029dcb09dfb5b2a2c783343efaf31a695bf648f/tests/integration/buildcmd/test_build_cmd.py#L1606)should cover that?